ucca/validation.py

```diff
@@ -122,6 +124,106 @@ def validate_non_terminal(self, linkage=False, multigraph=False):
                 if len(edges) > 1:
                     yield "Multiple edges from %s to %s: %s" % (self.node_id, child_id, ", ".join(
                         "%d %s" % (len(e), t) for t, e in tag_to_edge(edges).items()))
+        no_sub_non_scene = (ETags.Process, ETags.Adverbial, ETags.Linker, ETags.Time,
+                            ETags.Quantifier, ETags.Connector, ETags.State)
+        if any(k in self.incoming for k in no_sub_non_scene):
```

Comment:
This will not work. `self.incoming` is a list of edges, not strings (categories). Remember also that edges may have multiple categories, and their `tag` is what you want to check here. So you should use `incoming_tags`. And instead of the list comprehension you can use `intersection` with the `incoming_tags`.